spiketoolkit/sorters/kilosort/kilosort.py

```diff
@@ -155,26 +155,27 @@ def _setup_recording(self, recording, output_folder):
             groups = [recording.get_channel_property(ch, 'group') for ch in recording.get_channel_ids()]
         else:
             groups = 'ones(1, Nchannels)'
-        if 'location' in recording.get_channel_property_names():
-            positions = np.array([recording.get_channel_property(chan, 'location') for chan in recording.get_channel_ids()])
-            if electrode_dimensions is None:
-                kilosort_channelmap = ''.join(kilosort_channelmap
-                                              ).format(nchan,
-                                                       list(positions[:, 0]),
-                                                       list(positions[:, 1]),
-                                                       groups,
-                                                       sample_rate)
-            elif len(electrode_dimensions) == 2:
-                kilosort_channelmap = ''.join(kilosort_channelmap
-                                              ).format(nchan,
-                                                       list(positions[:, electrode_dimensions[0]]),
-                                                       list(positions[:, electrode_dimensions[1]]),
-                                                       groups,
-                                                       recording.get_sampling_frequency())
-            else:
-                raise Exception("Electrode dimension should bi a list of len 2")
+        if 'location' not in recording.get_channel_property_names():
+            print("'location' information is not found. Using linear configuration")
+            for i_ch, ch in enumerate(recording.get_channel_ids()):
+                recording.set_channel_property(ch, 'location', [0, i_ch])
+        positions = np.array([recording.get_channel_property(chan, 'location') for chan in recording.get_channel_ids()])
+        if electrode_dimensions is None:
+            kilosort_channelmap = ''.join(kilosort_channelmap
+                                          ).format(nchan,
+                                                   list(positions[:, 0]),
+                                                   list(positions[:, 1]),
+                                                   groups,
+                                                   sample_rate)
```

Comment:
This will make the generated file with MEArec not working because you use dim 1 and 2.
No ?
We shoucld write somewhere that if 3D in recording but sorter need 2D then the first 2 dimes are taken magically.

Reply:
you can use the electrode_dimension. if `electrode_dimension=[1,2]` then those dimensions are used

Reply:
we can deal with it in another PR :)

Reply:
Of course but in this futur PR, we will nede to detsroy this trick inside KS.
Lets make an issue to not forget it.